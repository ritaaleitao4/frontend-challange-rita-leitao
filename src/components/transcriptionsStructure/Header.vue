<template>
  <header>
    <div class="header-content">
      <h1>Transcriptions</h1>
      <aside>
        <uploadSvg class="upload" @click="uploadData"/>
        <fetch-svg class="fetch" @click="fetchData"/>
      </aside>
    </div>
  </header>
</template>

<script>
  import uploadSvg from '@/assets/images/upload.svg'
  import fetchSvg from '@/assets/images/fetch-document.svg'

  export default {
    name: 'Header',
    components: {
      uploadSvg,
      fetchSvg,
    },
    methods: {
      uploadData() {
        this.$emit('uploadData')
      },
      fetchData() {
        this.$emit('fetchData')
      },
    },
  }
</script>

<style scoped lang="scss">
  header {
    background-color: #FFFFFF;
    width: 100%;
    height: $default-header-height;
    box-shadow: 0 1px 2px 0 rgba(0,0,0,0.24), 0 1px 3px 0 rgba(0,0,0,0.12);
    display: flex;
    flex-flow: row nowrap;
    justify-content: center;

    .header-content {
      width: 100%;
      max-width: $default-header-width;
      padding: 0 $default-padding;
      display: flex;
      flex-flow: row nowrap;
      align-items: center;
      justify-content: space-between;

      h1 {
        @include header-title;
      }

      svg {
        width: 24px;
        height: 24px;
        cursor: pointer;
        margin-left: $default-margin;
        display: inline-block;
      }
    }
  }
</style>
