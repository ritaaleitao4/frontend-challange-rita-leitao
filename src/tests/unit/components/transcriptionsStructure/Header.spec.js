import Vue from 'vue'
import { createLocalVue, shallowMount } from '@vue/test-utils'
import Header from '@/components/transcriptionsStructure/Header.vue'

const localVue = createLocalVue()
const wrapper = shallowMount(Header, {
  localVue,
})

it('Snapshot test', () => {
  const Constructor = Vue.extend(Header)
  const vm = new Constructor().$mount()
  expect(vm.$el).toMatchSnapshot()
})

describe('Header Test', () => {
  it('should emit event to upload data', () => {
    wrapper.find('.upload').trigger('click')
    expect(wrapper.emitted('uploadData')).toBeTruthy()
  })

  it('should emit event to fetch data', () => {
    wrapper.find('.fetch').trigger('click')
    expect(wrapper.emitted('fetchData')).toBeTruthy()
  })
})
