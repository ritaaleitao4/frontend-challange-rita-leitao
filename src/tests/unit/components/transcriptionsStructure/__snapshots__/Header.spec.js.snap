// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
<header>
  <div
    class="header-content"
  >
    <h1>
      Transcriptions
    </h1>
     
    <aside>
      <upload-svg
        class="upload"
      />
       
      <fetch-svg
        class="fetch"
      />
    </aside>
  </div>
</header>
`;
